File: Pet-Adoption-Shelter-Front/src/apis/auth/auth.api.ts
import type { LoginRequest, LoginResponse, LogoutRequest, PasswordResetRequest, PasswordVerifyResponse, RefreshRequest, SignUpRequest, SignUpResponse } from "@/types/auth/auth.dto";
import { privateApi, publicApi } from "../common/axiosInstance";
import { AUTH_PATH } from "./auth.path";
import type { ApiResponse } from "@/types/common/ApiResponse";

export const authApi = {
  signUp: async (req: SignUpRequest): Promise<SignUpResponse> => {
    const res = await publicApi.post<ApiResponse<SignUpResponse>>(AUTH_PATH.SIGNUP, req);
    return res.data.data;
  },

  login: async (req: LoginRequest): Promise<LoginResponse> => {
    const res = await publicApi.
    post<ApiResponse<LoginResponse>>(
      AUTH_PATH.LOGIN, req
    );
    return res.data.data;
  },

  logout: async (req: LogoutRequest) => {
    await privateApi.post<void>(AUTH_PATH.LOGOUT, req);
  },

  refresh: async (req: RefreshRequest) => {
    await privateApi.post<void>(AUTH_PATH.REFRESH, req);
  },

  passwordReset: async (req: PasswordResetRequest) => {
    await privateApi.post<void>(AUTH_PATH.PASSWORD_RESET, req);
  },

  passwordVerify: async (): Promise<PasswordVerifyResponse> => {
    const res = await privateApi.get<ApiResponse<PasswordVerifyResponse>>(AUTH_PATH.VERIFY);
    return res.data.data;
  }
}

File: mini-proj-Pet-Adoption-Shelter/src/main/java/org/example/miniprojpetadoptionshelter/common/apis/AuthApi.java
package org.example.miniprojpetadoptionshelter.common.apis;

public class AuthApi {
    private AuthApi () {}

    public static final String ROOT = ApiBase.BASE + "/auth";
    public static final String SIGNUP = ROOT + "/signup";
    public static final String LOGIN = ROOT + "/login";
    public static final String LOGOUT = ROOT + "/logout";
    public static final String REFRESH = ROOT + "/refresh";
    public static final String PASSWORD_RESET = ROOT + "/password/reset";
    public static final String VERIFY = ROOT + "/verify";
}
